vizro-core/examples/visual-vocabulary/custom_charts.py

```diff
@@ -233,3 +233,70 @@ def dumbbell(data_frame: pd.DataFrame, x: str, y: str, color: str) -> go.Figure:
     # Increase size of dots
     fig.update_traces(marker_size=12)
     return fig
+
+
+@capture("graph")
+def diverging_stacked_bar(
+    data_frame,
+    y: str,
+    category_pos: List[str],
+    category_neg: List[str],
+    color_discrete_map: Optional[Dict[str, str]] = None,
+) -> go.Figure:
+    """Creates a horizontal diverging stacked bar chart (with positive and negative values only) using Plotly's go.Bar.
+
+    This type of chart is a variant of the standard stacked bar chart, with bars aligned on a central baseline to
+    show both positive and negative values. Each bar is segmented to represent different categories.
+
+    This function is not suitable for diverging stacked bar charts that include a neutral category.
+
+    Inspired by: https://community.plotly.com/t/need-help-in-making-diverging-stacked-bar-charts/34023
+
+    Args:
+       data_frame (pd.DataFrame): The data frame for the chart.
+       y (str): The name of the categorical column in the data frame to be used for the y-axis (categories)
+       category_pos (List[str]): List of column names in the data frame representing positive values. Columns should be
+            ordered from least to most positive.
+       category_neg (List[str]): List of column names in the DataFrame representing negative values. Columns should be
+            ordered from least to most negative.
```

Comment:
I don't understand why columns should be ordered from least to most positive (and similar to negative values), and also what exactly that means. When I changed the example configuration like below (I just switched the "category_pos" order of columns - and `color_discrete_map` additionally), I got the chart like this below: 

```py
                category_pos=["Strongly Agree", "Agree"],
                category_neg=["Disagree", "Strongly Disagree"],
                color_discrete_map={
                    "Agree": "#1a85ff",
                    "Strongly Agree": "#70a1ff",
                    "Disagree": "#ff5584",
                    "Strongly Disagree": "#d41159",
                },
```

<img width="769" alt="image" src="https://github.com/user-attachments/assets/2b6dd696-861d-4dd5-bcf7-48a194e38d4b">

Reply:
In a diverging stacked bar chart, it's really important to sort the colors in a specific order. This is because the chart draws the sections one after the other starting from the middle. In your example, the right side starts with "Strongly Agree" and then "Agree," which feels off because the Likert scale naturally goes from "Strongly Disagree" to "Disagree," then "Agree," and finally "Strongly Agree."

When the natural order isn't followed, like in your current setup, it makes the chart harder to read and understand. The colors aren't just random categories where the order doesn't matter; in a diverging stacked bar chart, the order is key. It always shows responses from least to most extreme on both the positive and negative sides.

So, it's important to arrange the categories from least extreme to most extreme to make sure the chart is clear and easy to interpret :)

Reply:
Yes, exactly. It's a data viz best practice guidance :) I don't want to enforce it via checks etc. , because in the end it's still their chart.